src/blog/views/views_v1.py

```diff
@@ -0,0 +1,191 @@
+        if serializer.is_valid(raise_exception=True):
+            serializer.save()
+            return Response(
+                    data=serializer.data,
+                    status=status.HTTP_201_CREATED
+                    )
+
+    def put(self, request, slug):
+        article = self.get_object(slug)
+        serializer = ArticleUpdateSerializer(data=request.data, instance=article)
+        if serializer.is_valid(raise_exception=True):
+            serializer.save()
+            return Response(
+                    data=serializer.data,
+                    status=status.HTTP_201_CREATED
+                    )
+
+    
+class MainCommentView(APIView):
+    def get(self, request, slug):
+        # Prefetch only the main comments (no parent)
+        main_comments = Prefetch('comments', queryset=CommentModel.objects.filter(parent__isnull=True))
+
+        # Retrieve the article with the prefetch of main comments and annotate comment count
+        article = get_object_or_404(
+                    ArticleModel.objects.prefetch_related(main_comments).annotate(
+                    comment_count=Count('comments', filter=Q(comments__parent__isnull=True))
+                    ),slug=slug)
+        
+        # Serialize the article with its main comments and comment count
+        serializer = ArticlesCommentSerializer(article)
+
+        return Response(
+            data=serializer.data,
+            status=status.HTTP_200_OK
+        )
+
+    
+class AllCommentsView(APIView):
+    def get(self, request, id):
+        # Retrieve the comment by its ID
+        comment = get_object_or_404(CommentModel, id=id)
+
+        print("comment", comment)
+        # Retrieve all replies to the comment 
+        replies = CommentModel.objects.filter(parent=comment)
+
+        # Serialize the replies
+        serializer = AllCommentSerializer(replies, many=True)
+
+        # Return the serialized data
+        return Response(
+            data=serializer.data,
+            status=status.HTTP_200_OK
+        )
+    
+
+class BookMarkView(APIView):
+    def post(self, request, article_id):
+        request_user = CustomUser.objects.get(id=1)
+
+        # Check if the user has already bookmarked this article
+        if BookMarkModel.objects.filter(user=request_user, article__id=article_id).exists():
+            return Response({"detail": "You have already bookmarked this article."}, status=status.HTTP_400_BAD_REQUEST)
+        
+        serializer = BookMarkSerializer(data=request.data)
+        if serializer.is_valid(raise_exception=True):
+            serializer.save(user=request_user, article_id=article_id)
+            return Response(
+                    data=serializer.data,
+                    status=status.HTTP_201_CREATED)
+    
+    
+    def delete(self, request, article_id):
+        article = get_object_or_404(BookMarkModel, article_id=article_id)
```

Comment:
# multiple users can  add same article as bookmark, so it must be 
`article = get_object_or_404(BookMarkModel, article_id=article_id,author(Idon't know exact name of field)=request.user)`